packages/hebao_v1/contracts/base/MetaTxModule.sol

```diff
@@ -131,6 +139,45 @@ contract MetaTxModule is BaseModule
         }
     }
 
+    /// @dev Helper method to execute multiple transactions as part of a single meta transaction.
+    ///      This method can only be called by a meta transaction.
+    /// @param wallet The wallet used in all transactions.
+    /// @param metaTxHash The hash of the complete meta transaction that is signed.
+    /// @param signatures The signatures used for all transactions. If transactions need different number
+    ///        of signatures the signatures with indices [0, numSigners[ are used.
+    /// @param data The raw transaction data used for each transaction.
+    /// @param value The ETH value to send in each transaction (total MUST match msg.value of meta tx).
+    function executeTransactions(
+        address            wallet,
+        bytes32            metaTxHash,
+        bytes[]   calldata signatures,
+        bytes[]   calldata data,
+        uint[]    calldata value
+        )
+        external
+        payable
+        onlyFromMetaTx
+    {
+        require(data.length == value.length, "INVALID_INPUT");
+        uint totalValue = 0;
+        for (uint i = 0; i < data.length; i++) {
+            // Check that the wallet is the same for all transactions
+            address txWallet = extractWalletAddress(data[i]);
+            require(txWallet == wallet, "INVALID_DATA");
+
+            // Verify the signatures for this transaction
+            bytes4 method = extractMethod(data[i]);
+            address[] memory signers = extractMetaTxSigners(wallet, method, data[i]);
+            metaTxHash.verifySignatures(signers, signatures);
```

Comment:
the `signers` and `signatures` won't match correctly. For example, if signatures are from address, A, B, and C. but singers can be A, and C.

Reply:
Yeah, but I think having special ordering for signers will be pretty rare I think? Well, the wallet order at signers[0] is pretty common, but otherwise it doesn't seem common that signers need a fixed order.

I will try to find a way to make it as general as possible, but having to check all the signatures for each transaction (or needing complex logic to work around it) seems like a waste of gas for some very uncommon uses. But, I'm still trying to find better ways to do it.

Reply:
This is now fixed in an acceptable/cheap way I think. No order of signers is enforced any more.